core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -1118,14 +1118,16 @@ protected boolean parentExists(String path) throws IOException {
    */
   @VisibleForTesting
   public String stripPrefixIfPresent(String path) {
-    final String normalizedPath = PathUtils.normalizePath(path, PATH_SEPARATOR);
+    if (mRootKeySupplier.get().equals(path)) {
```

Comment:
mRootKeySupplier.get() is a memoized function so don't be concerned about callng twice.